app/bot.go

```diff
@@ -193,6 +193,93 @@ func (a *App) disableUserBots(userId string) *model.AppError {
 	return nil
 }
 
+func (a *App) notifySysadminsBotDisabled(userId string) *model.AppError {
+
+	disableBotsSetting := *a.Config().ServiceSettings.DisableBotsWhenOwnerIsDeactivated
+
+	perPage := 20
+	options := &model.BotGetOptions{
+		OwnerId:        userId,
+		IncludeDeleted: false,
+		OnlyOrphaned:   false,
+		Page:           0,
+		PerPage:        perPage,
+	}
+	userBots, err := a.GetBots(options)
+	if err != nil {
+		return err
+	}
+
+	// user does not own bots
+	if len(userBots) == 0 {
+		return nil
+	}
+
+	// user being disabled
+	user, err := a.GetUser(userId)
+	if err != nil {
+		return err
+	}
+
+	// get sysadmins
+	userOptions := &model.UserGetOptions{
+		Page:     0,
+		PerPage:  perPage,
+		Role:     model.SYSTEM_ADMIN_ROLE_ID,
+		Inactive: false,
+	}
+	sysAdmins, err := a.GetUsers(userOptions)
+	if err != nil {
+		return err
+	}
+
+	// for each sysadmin, notify user that owns bots was disabled
+	for _, sysAdmin := range sysAdmins {
+		channel, appErr := a.GetOrCreateDirectChannel(sysAdmin.Id, sysAdmin.Id)
+		if appErr != nil {
+			return appErr
+		}
+
+		post := &model.Post{
+			UserId:    sysAdmin.Id,
+			ChannelId: channel.Id,
+			Message:   GetDisableBotSysadminMessage(user, userBots, disableBotsSetting),
+			Type:      model.POST_SYSTEM_GENERIC,
+		}
+
+		_, appErr = a.CreatePost(post, channel, false)
+		if appErr != nil {
+			return appErr
+		}
+	}
+	return nil
+}
+
+func GetDisableBotSysadminMessage(user *model.User, userBots model.BotList, disableBotsSetting bool) string {
+
+	infoMessage := fmt.Sprintf("For more information, see our [documentation](https://docs.mattermost.com/developer/bot-accounts.html#what-happens-when-a-user-who-owns-bot-accounts-is-disabled).")
```

Comment:
This all needs to be translatable. The translations support templates.

Reply:
Translations added in the latest commits.